theme/assets/js/templates/store/item.js

```diff
@@ -495,6 +497,104 @@ document.addEventListener('alpine:init', () => {
         },
     });
 
+    const createAfterpayBadgeData = () => ({
+        isLoading: true,
+        payments: null,
+        paymentAmount: null,
+        selector: '.afterpay-placement',
+        /**
+         * Initialize Square Web Payments SDK
+         *
+         * @param {String} applicationId the application id
+         * @param {String} locationId the location id
+         * @return {Object}
+        */
+        initWebPaymentsSDK(applicationId, locationId) {
+            if (!this.webPaymentsSDKLoaded()) {
+                console.error('Web Payments SDK not loaded');
+            }
+
+            const Square = window.Square;
+            return Square.payments(applicationId, locationId);
+        },
+
+        /**
+         * Whether the Web Payments SDK has already been loaded
+         * @return {Boolean}
+         */
+        webPaymentsSDKLoaded() {
+            return !!window.Square;
+        },
+        async init() {
+            // Initialize the payments SDK
+            const applicationId = Alpine.store('global').applicationId;
+            const locationId = Alpine.store('product').locationId;
+            const storeLocale = Alpine.store('global').locale;
+            const payments = this.initWebPaymentsSDK(applicationId, locationId);
+            payments.setLocale(storeLocale);
+            this.payments = payments;
+
+            this.paymentAmount = this.product.price.regular_low.amount;
+
+            // Load afterpay messaging
+            await this.attachAfterpayMessaging();
+            this.$watch('$store.product.selectedVariationPrice', () => this.reloadAfterpayMessaging());
+        },
+        reloadAfterpayMessaging() {
+            this.cleanupAfterpayMessaging();
+            this.paymentAmount = Alpine.store('product').selectedVariationPrice;
+            this.attachAfterpayMessaging();
+        },
+        async attachAfterpayMessaging() {
+            this.isLoading = true;
+            this.error = null;
+
+            try {
+                const paymentRequest = this.buildPaymentRequest(this.paymentAmount);
+                const afterpay = await this.payments.afterpayClearpay(paymentRequest);
+                this.cleanupAfterpayMessaging();
+                await afterpay.attachMessaging(this.selector, Constants.AFTERPAY_MESSAGING_OPTIONS);
+                this.isLoading = false;
+            } catch (err) {
```

Comment:
i see we have `this.error` defined above, should this set `this.error = true` when it fails?